apps/redi-connect/src/pages/app/applications/DesktopView.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+
+import { Button } from '@talent-connect/shared-atomic-design-components'
+import ApplicationCard from '../../../components/organisms/ApplicationCard'
+import './DesktopView.scss'
+
+const DesktopView = ({
+  applicants,
+  filteredApplicants,
+  activeFilter,
+  setActiveFilter,
+}) => {
+  const pendingApplications = applicants.filter(
+    (applicant) => applicant.status === 'applied'
+  )
+  const hasPendingApplications = Boolean(pendingApplications.length)
+
+  const hasAcceptedApplications = applicants.some(
+    (applicant) =>
+      applicant.status === 'accepted' || applicant.status === 'completed'
+  )
+
+  const hasDeclinedApplications = applicants.some(
+    (applicant) => applicant.status === 'declined-by-mentor'
+  )
+
+  const hasCancelledApplications = applicants.some(
+    (applicant) =>
+      applicant.status === 'cancelled' ||
+      applicant.status === 'invalidated-as-other-mentor-accepted'
+  )
+
+  const handleActiveFilter = (tabName) => {
+    setActiveFilter(tabName)
+  }
+
+  const getClassName = (tabName) => {
+    return activeFilter === tabName
+      ? 'tabs-menu__item--active'
+      : 'tabs-menu__item'
+  }
+
+  return (
+    <div className="desktop-tabs">
+      <div className="tabs-menu">
+        <Button
```

Comment:
Decent use case for applying DRY. Can you refactor so the client code looks something like this?
```
<FilterButton isActive={isActive('all')} handleFilterSelect={handleActiveFilter('all')}>
   all {Boolean(applicants.length) && `(${applicants.length})`}
</FilterButton>
<FilterButton isActive={isActive('pending')} handleFilterSelect={handleActiveFilter('pending')}>
   pending {hasPendingApplications && `(${pendingApplications.length})`}
</FilterButton>
... and so forth
```

Reply:
As we discussed, I refactored by adding Context for an active filter.